olp-cpp-sdk-dataservice-read/tests/StreamApiTest.cpp

```diff
@@ -0,0 +1,183 @@
+
+    OlpClientSettings settings;
+    settings.network_request_handler = network_mock_;
+    settings.task_scheduler =
+        OlpClientSettingsFactory::CreateDefaultTaskScheduler(1);
+    olp_client_.SetSettings(std::move(settings));
+  }
+
+  void TearDown() override { network_mock_.reset(); }
+
+ protected:
+  OlpClient olp_client_;
+  std::shared_ptr<NetworkMock> network_mock_;
+};
+
+const std::string kBaseUrl{
+    "https://some.base.url/stream/v2/catalogs/"
+    "hrn:here:data::olp-here-test:hereos-internal-test-v2"};
+const std::string kNodeBaseUrl{
+    "https://some.node.base.url/stream/v2/catalogs/"
+    "hrn:here:data::olp-here-test:hereos-internal-test-v2"};
+const std::string kSubscriptionId{"test-subscription-id-123"};
+const std::string kConsumerId{"test-consumer-id-987"};
+const std::string kLayerId{"test-layer"};
+const std::string kCorrelationId{"test-correlation-id"};
+const std::string kSerialMode{"serial"};
+const std::string kParallelMode{"parallel"};
+
+constexpr auto kUrlStreamSubscribeNoQueryParams =
+    R"(https://some.base.url/stream/v2/catalogs/hrn:here:data::olp-here-test:hereos-internal-test-v2/layers/test-layer/subscribe)";
+
+constexpr auto kUrlStreamSubscribeWithQueryParams =
+    R"(https://some.base.url/stream/v2/catalogs/hrn:here:data::olp-here-test:hereos-internal-test-v2/layers/test-layer/subscribe?consumerId=test-consumer-id-987&mode=serial&subscriptionId=test-subscription-id-123)";
+
+constexpr auto kHttpResponseSubscribeSucceeds =
+    R"jsonString({ "nodeBaseURL": "https://some.node.base.url/stream/v2/catalogs/hrn:here:data::olp-here-test:hereos-internal-test-v2", "subscriptionId": "test-subscription-id-123" })jsonString";
+
+constexpr auto kHttpResponseSubscribeFails =
+    R"jsonString({ "title": "Subscription mode not supported", "status": 400, "code": "E213002", "cause": "Subscription mode 'singleton' not supported", "action": "Retry with valid subscription mode 'serial' or 'parallel'", "correlationId": "4199533b-6290-41db-8d79-edf4f4019a74" })jsonString";
+
+constexpr auto kHttpRequestBodyWithConsumerProperties =
+    R"jsonString({"kafkaConsumerProperties":{"field_string":"abc","field_int":"456","field_bool":"1"}})jsonString";
+
+TEST_F(StreamApiTest, Subscribe) {
+  {
+    SCOPED_TRACE("Subscribe without optional input fields succeeds");
+
+    EXPECT_CALL(
+        *network_mock_,
+        Send(IsPostRequest(kUrlStreamSubscribeNoQueryParams), _, _, _, _))
+        .WillOnce(ReturnHttpResponse(
+            http::NetworkResponse().WithStatus(http::HttpStatusCode::CREATED),
+            kHttpResponseSubscribeSucceeds));
+
+    olp_client_.SetBaseUrl(kBaseUrl);
+    std::string x_correlation_id;
+    CancellationContext context;
+    const auto subscribe_response = StreamApi::Subscribe(
+        olp_client_, kLayerId, boost::none, boost::none, boost::none,
+        boost::none, context, x_correlation_id);
+
+    ASSERT_TRUE(subscribe_response.IsSuccessful())
```

Comment:
```suggestion
    EXPECT_TRUE(subscribe_response.IsSuccessful())
```
I think you can use `EXPECT` here instead assert in order to not block other SCOPED_TRACE from invocation.